src/common/components/SubscribeChannelDropdown.tsx

```diff
@@ -28,42 +28,35 @@ export type SubscribeChannelDropdownProps = {
 const SubscribeChannelDropdown: FC<SubscribeChannelDropdownProps> = (options) => {
   const { children, channelDetails, onSuccess } = options;
   const { account, provider, wallet, chainId } = useAccount();
-  const { connectWallet } = useContext(AppContext);
 
-  const channelSettings = useMemo(() => {
-    if (channelDetails && channelDetails?.channel_settings) {
-      return JSON.parse(channelDetails?.channel_settings);
-    }
-    return null;
-  }, [channelDetails]);
+  const context = useContext(AppContext);
+  if (!context) {
+    throw new Error('App Context is null');
+  }
+  const { connectWallet } = context;
+
```

Comment:
Not the correct approach. Please fix this. Create a custom hook for the context and add relevant there.